sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -212,7 +224,10 @@ object V2ScanRelationPushDown extends Rule[LogicalPlan] with PredicateHelper {
 
   private def supportPartialAggPushDown(agg: Aggregation): Boolean = {
     // We don't know the agg buffer of `GeneralAggregateFunc`, so can't do partial agg push down.
-    agg.aggregateExpressions().forall(!_.isInstanceOf[GeneralAggregateFunc])
+    agg.aggregateExpressions().forall { aggregateFunc =>
+      !aggregateFunc.isInstanceOf[GeneralAggregateFunc] ||
+        aggregateFunc.asInstanceOf[GeneralAggregateFunc].name() == "AVG"
```

Comment:
If `AVG` can be partially pushed, I'd like to create a dedicated class for it, instead of using `GeneralAggregateFunc`.